runtime/compiler/runtime/RuntimeAssumptions.hpp

```diff
@@ -142,6 +142,9 @@ class TR_PersistentClassInfo : public TR_Link0<TR_PersistentClassInfo>
    bool shouldNotBeNewlyExtended(int32_t ID)         { return _shouldNotBeNewlyExtended.testAny(1 << ID); }
    flags16_t getShouldNotBeNewlyExtendedMask() const { return _shouldNotBeNewlyExtended; }
 
+   void setAlreadyScannedForFinalPutstatic() { _flags.set(_alreadyScannedForFinalPutstatic, true); }
```

Comment:
We may need to redefine `setAlreadyScannedForFinalPutstatic()` in TR_JITClientPersistentClassInfo much like we do for `setCannotTrustStaticFinal()`. This ensures that modifications at the client are sent to the server.
It's possible that some compilations are performed at the client (either because of heuristics or because of temporary loss of connection) and the flag `_alreadyScannedForFinalPutstatic` is only set at the client. It's not a correctness issue because the server will just need to scan the class again, but I would still have this change performed.